sevenn/pair_e3gnn/README.md

```diff
@@ -0,0 +1,133 @@
+We support the use of the Grimme's D3 dispersion (van der Waals) correction scheme accelerated by CUDA, which can be used within LAMMPS in conjunction with SevenNet.
+
+**PLEASE NOTE:** Currently, this D3 code does not support GPU parallelism yet. So it can only be run on a single GPU.
+
+# About Grimme's D3 code accelerated with CUDA 
+
+This is LAMMPS implementation of [Grimme's D3 method](https://doi.org/10.1063/1.3382344). We have ported the code from the [original fortran code](https://www.chemie.uni-bonn.de/grimme/de/software/dft-d3) to a LAMMPS pair style written in C++. The D3 method is semi-empirical and significantly faster than DFT, but it runs slower when combined with SevenNet. Therefore, we have adopted CUDA to accelerate the code.
+
+We tested various implementations using MPI, OpenMP, OpenACC, and CUDA to accelerate the code. Among these, OpenACC and CUDA proved to be the most efficient. However, since OpenACC is not compatible with libtorch, we chose to use the CUDA version.
+
+Initially, we implemented the code using double precision (FP64), similar to the original, but to further speed up the calculations, we transitioned some operations to single precision (FP32). Consequently, this code may introduce minor numerical errors, particularly noticeable in stress calculations. However, these errors are significantly smaller than the numerical errors inherent in SevenNet itself, so we have decided to use FP32-based operations.
+
+# Installation of D3 code for LAMMPS
+
+We support the installation of D3 code with SevenNet via the `sevenn_patch_lammps` command. You can simply add the `--d3` argument:
+
+```bash
+sevenn_patch_lammps ./lammps_sevenn --d3
+```
+
+You can follow the remaining installation steps in the SevenNet documentation. For detailed installation options, refer to `sevenn/pair_e3gnn/patch_lammps.sh`.
+
+**PLEASE NOTE**: In this code, the selection of compute capability for GPUs relies on FindCUDA in libtorch. However, the default CUDA architectures selected in CUDA 12.1 are `"50;80;86;89;90"`. The D3 code, however, requires at least compute capability 6.0. If you compile with the default settings, you may encounter an `atomicAdd` error. To resolve this issue, I have included code in the script to remove the compute capability for 5.0.
```

Comment:
Supporting GPU architectures (defaults are marked with *): ~~
You can find your GPU architecture from 'here'